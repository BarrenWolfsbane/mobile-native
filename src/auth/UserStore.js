import {
  observable,
  action
} from 'mobx';

import channelService from './../channel/ChannelService';
import sessionService from '../common/services/session.service';
import UserModel from './../channel/UserModel';
import { MINDS_FEATURES } from '../config/Config';
import { Alert } from 'react-native';

/**
 * Login Store
 */
class UserStore {
  @observable me = {};

  @action
  setUser(user) {
    if (!user || !user.guid)
      return
    this.me = UserModel.create(user);
  }

  @action
  setRewards(value) {
    this.me.rewards = !!value;
  }

  @action
  setWallet(value) {
    this.me.eth_wallet = value;
  }

  @action
  setTosLastUpdate(value) {
    this.me.last_accepted_tos = value;
  }

  hasRewards() {
    return !!(this.me && this.me.rewards);
  }

  hasEthWallet() {
    return !!(this.me && this.me.eth_wallet);
  }

  hasAvatar() {
    return this.me.icontime !== this.me.time_created;
  }

  @action
  clearUser() {
    this.me = {};
  }

  @action
  async load(refresh = false) {
    if (!refresh) this.me = {};

    let response = await channelService.load('me');

    //if (!response.channel) {
    //  return sessionService.setToken(null);
    //}

    this.setUser(response.channel);
    if (this.me.canCrypto) {
      MINDS_FEATURES.crypto = true;
    }
    return this.me;

  }

  isAdmin() {
    return this.me.admin;
  }

  @action
  reset() {
    this.me = {};
  }

}

export default UserStore;
