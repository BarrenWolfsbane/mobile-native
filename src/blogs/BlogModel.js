import { decorate, observable } from 'mobx';
import { MINDS_ASSETS_CDN_URI } from '../config/Config';
import api from '../common/services/api.service';
import BaseModel from '../common/BaseModel';
import UserModel from '../channel/UserModel';
import { LICENSES } from '../common/services/list-options.service';

/**
 * User model
 */
export default class BlogModel extends BaseModel {

  /**
   * Get banner source
   * @param {string} size
   */
  getBannerSource(size='medium') {
    const uri = (this.thumbnail_src && this.thumbnail_src != 'https://cdn.minds.com/thumbProxy?src=&c=2708')?
      this.thumbnail_src:
      `${MINDS_ASSETS_CDN_URI}front/dist/assets/logos/placeholder-bulb.jpg`;

    return { uri, headers: api.buildHeaders()};
  }

  /**
   * Child models
   */
  childModels() {
    return {
      ownerObj: UserModel
    }
  }

  /**
   * Get the license text
   */
  getLicenseText() {
    const lic =  LICENSES.find(license => this.license == license.value);
    if (!lic) return;
    return lic.text;
  }
}

/**
 * Define model observables
 */
decorate(BlogModel, {
  'thumbs:down:count': observable,
  'thumbs:up:count': observable,
  'thumbs:down:user_guids': observable,
  'thumbs:up:user_guids': observable,
});