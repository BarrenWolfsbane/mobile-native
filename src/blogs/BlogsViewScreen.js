import React, {
  Component
} from 'react';

import {
  View,
  StyleSheet,
  ScrollView,
  Dimensions,
  Image,
  TextInput,
  Keyboard,
  TouchableOpacity,
  TouchableHighlight,
  FlatList,
  Text,
  Platform,
  KeyboardAvoidingView,
  ActivityIndicator,
} from 'react-native';

import { inject, observer } from 'mobx-react/native';

import FastImage from 'react-native-fast-image';
import { Icon } from 'react-native-elements';
import IonIcon from 'react-native-vector-icons/Ionicons';

import BlogViewHTML from './BlogViewHTML';
import OwnerBlock from '../newsfeed/activity/OwnerBlock';
import formatDate from '../common/helpers/date';
import { CommonStyle as CS } from '../styles/Common';
import colors from '../styles/Colors';
import ThumbUpAction from '../newsfeed/activity/actions/ThumbUpAction';
import ThumbDownAction from '../newsfeed/activity/actions/ThumbDownAction';
import RemindAction from '../newsfeed/activity/actions/RemindAction';
import CommentsAction from '../newsfeed/activity/actions/CommentsAction';
import shareService from '../share/ShareService';
import commentsStoreProvider from '../comments/CommentsStoreProvider';
import CommentList from '../comments/CommentList';
import CenteredLoading from '../common/components/CenteredLoading';

/**
 * Blog View Screen
 */
@inject('user', 'blogsView')
@observer
export default class BlogsViewScreen extends Component {

  /**
   * Disable navigation bar
   */
  static navigationOptions = {
    header: null
  }

  share = () => {
    const blog = this.props.blogsView.blog;
    shareService.share(blog.title, blog.perma_url);
  }

  state = {
    error: null
  };

  /**
   * Constructor
   * @param {object} props
   */
  constructor(props) {
    super(props);

    const params = props.navigation.state.params;
    this.comments = commentsStoreProvider.get();
  }

  /**
   * Component did mount
   */
  async componentDidMount() {
    const params = this.props.navigation.state.params;

    if (params.blog) {
      this.props.blogsView.setBlog(params.blog);
    } else {
      this.props.blogsView.reset();
      let guid;
      if (params.slug) {
        guid = params.slug.substr(params.slug.lastIndexOf('-')+1);
      } else {
        guid = params.guid;
      }
      try {
        await this.props.blogsView.loadBlog(guid);
      } catch (e) {
        this.setState({error: e});
      }
    }
  }

  /**
   * Render blog
   */
  getHeader() {
    const blog = this.props.blogsView.blog;

    const actions = (
      <View style={[CS.flexContainer, CS.paddingLeft2x]}>
        <View style={styles.actionsContainer}>
          <RemindAction entity={blog} size={16} />
          <ThumbUpAction entity={blog} orientation='column' size={16} me={this.props.user.me} />
          <ThumbDownAction entity={blog} orientation='column' size={16} me={this.props.user.me} />
        </View>
      </View>
    )
    const image = blog.getBannerSource();
    return (
      <View style={styles.screen}>
        <FastImage source={image} resizeMode={FastImage.resizeMode.cover} style={styles.image} />
        <Text style={styles.title}>{blog.title}</Text>
        <View style={styles.ownerBlockContainer}>
          <OwnerBlock entity={blog} navigation={this.props.navigation} rightToolbar={actions}>
            <Text style={styles.timestamp}>{formatDate(blog.time_created)}</Text>
          </OwnerBlock>
        </View>
        <View style={styles.description}>
          <BlogViewHTML html={blog.description} />
        </View>
        <View style={styles.moreInformation}>
          <Icon color={colors.medium} size={18} name='public' onPress={() => this.props.navigation.goBack()} />
          <Text style={[CS.fontXS, CS.paddingLeft, CS.colorMedium, CS.paddingRight2x]}>{blog.getLicenseText()}</Text>
          <Icon color={colors.primary} size={20} name='share' onPress={this.share} />
        </View>
        <Icon raised color={colors.primary} containerStyle={styles.header} size={30} name='arrow-back' onPress={() => this.props.navigation.goBack()}/>
        { this.comments.loadPrevious && !this.comments.loading ?
            <TouchableHighlight
            onPress={() => { this.loadComments()}}
            underlayColor = 'transparent'
            style = {styles.loadCommentsContainer}
          >
            <Text style={styles.loadCommentsText}> LOAD EARLIER </Text>
          </TouchableHighlight> : null
        }
      </View>
    )
  }

  /**
   * Render
   */
  render() {

    if (!this.props.blogsView.blog) return <CenteredLoading />;
    return (
      <View style={[CS.flexContainer, CS.backgroundWhite]}>
        {
          !this.state.error ?
            <CommentList
              header={this.getHeader()}
              entity={this.props.blogsView.blog}
              store={this.comments}
              navigation={this.props.navigation}
            />
          :
            <View style={CS.flexColumnCentered}>
              <FastImage
                resizeMode={FastImage.resizeMode.contain}
                style={ComponentsStyle.logo}
                source={require('../assets/logos/logo.png')}
              />
              <Text style={[CS.fontL, CS.colorDanger]}>SORRY, WE COULDN'T LOAD THE BLOG</Text>
              <Text style={[CS.fontM]}>PLEASE TRY AGAIN LATER</Text>
            </View>
        }
      </View>
    );
  }
}

let paddingBottom = 0;

const d = Dimensions.get('window');
if (d.height == 812 || d.width == 812) {
  paddingBottom = 16;
}

/**
 * Styles
 */
const styles = StyleSheet.create({
  containerContainer: {
    flex: 1,
    paddingBottom: paddingBottom,
  },
  header: {
    position: 'absolute',
    left: 0,
    top: 0,
    height: 40,
    width: 40,
  },
  actionsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: 4
  },
  title: {
    paddingTop: 12,
    paddingBottom: 8,
    paddingLeft: 12,
    paddingRight: 12,
    fontSize: 22,
    color: '#444',
    fontFamily: 'Roboto',
    fontWeight: '800',
  },
  ownerBlockContainer: {
    margin: 8,
  },
  description: {
    paddingLeft: 12,
    paddingRight: 12,
    paddingBottom: 12,
  },
  screen: {
    backgroundColor: '#FFF',
    flex:1
  },
  image: {
    height: 200
  },
  timestamp: {
    fontSize: 11,
    color: '#888',
  },
  moreInformation: {
    padding: 12,
    flexDirection: 'row',
  },
  messagePoster: {
    flexDirection: 'row',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'baseline',
    backgroundColor: '#FFF',
    padding: 5
  },
  posterAvatar: {
    height: 36,
    width: 36,
    borderRadius: 18,
    borderWidth: StyleSheet.hairlineWidth,
    borderColor: '#EEE',
  },
  input: {
    marginLeft: 8,
  },
  preview: {
    height: 200,
    flexDirection: 'row',
    alignItems: 'stretch',
    position: 'relative',
  },
  deleteAttachment: {
    position: 'absolute',
    right: 8,
    top: 0,
    color: '#FFF'
  },
  sendicon: {
    paddingRight: 8
  },
  loadCommentsContainer: {
    backgroundColor: '#EEE',
    borderRadius: 3,
    justifyContent: 'center',
    flexDirection: 'row',
    padding: 8,
    margin: 8,
  },
  loadCommentsText: {
    color: '#888',
    fontSize: 10,
  },
});