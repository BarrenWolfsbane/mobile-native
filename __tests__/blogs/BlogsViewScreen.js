import 'react-native';
import React from 'react';
import BlogsViewScreen from '../../src/blogs/BlogsViewScreen';
import BlogsViewStore from '../../src/blogs/BlogsViewStore';
import blogsFaker from '../../__mocks__/fake/blogs/BlogFactory';
import BlogModel from '../../src/blogs/BlogModel';
import UserStore from '../../src/auth/UserStore';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('../../src/blogs/BlogsViewStore');
jest.mock('../../src/auth/UserStore');
jest.mock('../../src/blogs/BlogViewHTML', () => 'BlogViewHTML');
jest.mock('../../src/newsfeed/activity/actions/ThumbUpAction', () => 'ThumbUpAction');
jest.mock('../../src/newsfeed/activity/actions/ThumbDownAction', () => 'ThumbDownAction');
jest.mock('../../src/newsfeed/activity/actions/RemindAction', () => 'RemindAction');
jest.mock('../../src/newsfeed/activity/actions/CommentsAction', () => 'CommentsAction');

/**
 * Tests
 */
describe('blog view screen component', () => {

  let store, user;

  beforeEach(() => {
    store = new BlogsViewStore();
    user = new UserStore();
    user.me = {
      guid: 'guidguid'
    };
  });

  it('should renders correctly', () => {

    store.blog = BlogModel.create(blogsFaker('1'));

    navigation = {state: {params : {guid:1}}};

    const component = renderer.create(
      <BlogsViewScreen.wrappedComponent blogsView={store} navigation={navigation} user={user}/>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should load the blog by guid', () => {

    store.blog = BlogModel.create(blogsFaker('1'));

    navigation = {state: {params : {guid: 1}}};

    const component = renderer.create(

      <BlogsViewScreen.wrappedComponent blogsView={store} navigation={navigation} user={user}/>
    );

    expect(store.loadBlog).toBeCalledWith(1);
  });

  it('should set the blog from params', () => {

    store.blog = BlogModel.create(blogsFaker('1'));

    navigation = {state: {params : {blog: store.blog}}};

    const component = renderer.create(
      <BlogsViewScreen.wrappedComponent blogsView={store} navigation={navigation} user={user}/>
    );

    expect(store.setBlog).toBeCalledWith(store.blog);
  });
});