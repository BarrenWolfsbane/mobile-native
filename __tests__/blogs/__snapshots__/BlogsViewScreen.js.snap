// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog view screen component should renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <CommentList
    entity={
      BlogModel {
        "access_id": "2",
        "boost_rejection_reason": -1,
        "categories": Array [],
        "category": "education",
        "container_guid": "100",
        "custom_meta": Object {
          "author": "",
          "description": "",
          "title": "",
        },
        "description": "Some Blog",
        "draft_access_id": "0",
        "excerpt": "Some description",
        "featured": false,
        "featured_id": false,
        "guid": "1",
        "header_bg": "1",
        "header_top": "0",
        "last_save": "1524843907",
        "last_updated": "1524838665",
        "license": "attribution-noncommercial-cc",
        "mature": false,
        "monetized": false,
        "ownerObj": UserModel {
          "access_id": "2",
          "banned": "no",
          "blocked": undefined,
          "boostProPlus": false,
          "boost_autorotate": true,
          "boost_rating": "2",
          "briefdescription": "",
          "categories": Array [
            "music",
            "nature",
            "news",
            "politics",
            "science",
            "technology",
            "education",
            "art",
            "health",
          ],
          "chat": true,
          "city": "",
          "container_guid": "0",
          "deleted": "0",
          "disabled_boost": false,
          "dob": "",
          "eth_wallet": "0x6fffffdadae350ddd5a1777d5a1777a4f39f0317",
          "fb": false,
          "feature_flags": false,
          "featured_id": "725165605782560768",
          "founder": false,
          "gender": "private",
          "getAvatarSource": [MockFunction],
          "guid": 100,
          "icontime": "1523515420",
          "language": "en",
          "legacy_guid": false,
          "mature": "1",
          "mature_channel": "0",
          "mature_visibility": false,
          "merchant": Object {
            "exclusive": Object {
              "amount": "10",
              "enabled": true,
            },
            "id": "acct_Dsdfjdlsaf34243",
            "service": "stripe",
          },
          "monetized": "",
          "name": "Some User",
          "owner_guid": "0",
          "pinned_posts": Array [],
          "plus": true,
          "programs": Array [
            "affiliate",
          ],
          "rating": "1",
          "rewards": true,
          "signup_method": false,
          "site_guid": false,
          "social_profiles": Array [],
          "spam": "0",
          "subscribed": false,
          "subscriber": true,
          "subtype": false,
          "time_created": "1468113204",
          "time_updated": false,
          "type": "user",
          "username": "someusermane",
          "verified": true,
          "website": "",
          "wire_rewards": Object {
            "description": "",
            "rewards": Object {
              "money": Array [],
              "points": Array [],
            },
          },
        },
        "owner_guid": "1",
        "paywall": false,
        "perma_url": "https://www.minds.com/blog/view/836607056186159104",
        "published": true,
        "rating": 1,
        "reminds": 58,
        "route": "blog/view/836607056186159104",
        "slug": "",
        "subtype": "blog",
        "thumbnail_src": "https://www.minds.com/fs/v1/banners/836607056186159104/1524838665",
        "thumbs:down:count": 1,
        "thumbs:down:user_guids": Array [
          "3",
        ],
        "thumbs:up:count": 2,
        "thumbs:up:user_guids": Array [
          "1",
          "2",
        ],
        "time_created": "1524838665",
        "time_published": "1524838665",
        "time_updated": "1524862542",
        "title": "Some Title",
        "type": "object",
        "wire_threshold": "",
      }
    }
    header={
      <View
        style={
          Object {
            "backgroundColor": "#FFF",
            "flex": 1,
          }
        }
      >
        <ForwardRef
          resizeMode="cover"
          source={
            Object {
              "headers": Object {
                "App-Version": "3.7.0-rc1",
                "Cache-Control": "no-cache, no-store, must-revalidate",
                "Pragma": "no-cache",
              },
              "uri": "https://www.minds.com/fs/v1/banners/836607056186159104/1524838665",
            }
          }
          style={
            Object {
              "height": 200,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#444",
              "fontFamily": "Roboto",
              "fontSize": 22,
              "fontWeight": "800",
              "paddingBottom": 8,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            }
          }
        >
          Some Title
        </Text>
        <View
          style={
            Object {
              "margin": 8,
            }
          }
        >
          <OwnerBlock
            entity={
              BlogModel {
                "access_id": "2",
                "boost_rejection_reason": -1,
                "categories": Array [],
                "category": "education",
                "container_guid": "100",
                "custom_meta": Object {
                  "author": "",
                  "description": "",
                  "title": "",
                },
                "description": "Some Blog",
                "draft_access_id": "0",
                "excerpt": "Some description",
                "featured": false,
                "featured_id": false,
                "guid": "1",
                "header_bg": "1",
                "header_top": "0",
                "last_save": "1524843907",
                "last_updated": "1524838665",
                "license": "attribution-noncommercial-cc",
                "mature": false,
                "monetized": false,
                "ownerObj": UserModel {
                  "access_id": "2",
                  "banned": "no",
                  "blocked": undefined,
                  "boostProPlus": false,
                  "boost_autorotate": true,
                  "boost_rating": "2",
                  "briefdescription": "",
                  "categories": Array [
                    "music",
                    "nature",
                    "news",
                    "politics",
                    "science",
                    "technology",
                    "education",
                    "art",
                    "health",
                  ],
                  "chat": true,
                  "city": "",
                  "container_guid": "0",
                  "deleted": "0",
                  "disabled_boost": false,
                  "dob": "",
                  "eth_wallet": "0x6fffffdadae350ddd5a1777d5a1777a4f39f0317",
                  "fb": false,
                  "feature_flags": false,
                  "featured_id": "725165605782560768",
                  "founder": false,
                  "gender": "private",
                  "getAvatarSource": [MockFunction],
                  "guid": 100,
                  "icontime": "1523515420",
                  "language": "en",
                  "legacy_guid": false,
                  "mature": "1",
                  "mature_channel": "0",
                  "mature_visibility": false,
                  "merchant": Object {
                    "exclusive": Object {
                      "amount": "10",
                      "enabled": true,
                    },
                    "id": "acct_Dsdfjdlsaf34243",
                    "service": "stripe",
                  },
                  "monetized": "",
                  "name": "Some User",
                  "owner_guid": "0",
                  "pinned_posts": Array [],
                  "plus": true,
                  "programs": Array [
                    "affiliate",
                  ],
                  "rating": "1",
                  "rewards": true,
                  "signup_method": false,
                  "site_guid": false,
                  "social_profiles": Array [],
                  "spam": "0",
                  "subscribed": false,
                  "subscriber": true,
                  "subtype": false,
                  "time_created": "1468113204",
                  "time_updated": false,
                  "type": "user",
                  "username": "someusermane",
                  "verified": true,
                  "website": "",
                  "wire_rewards": Object {
                    "description": "",
                    "rewards": Object {
                      "money": Array [],
                      "points": Array [],
                    },
                  },
                },
                "owner_guid": "1",
                "paywall": false,
                "perma_url": "https://www.minds.com/blog/view/836607056186159104",
                "published": true,
                "rating": 1,
                "reminds": 58,
                "route": "blog/view/836607056186159104",
                "slug": "",
                "subtype": "blog",
                "thumbnail_src": "https://www.minds.com/fs/v1/banners/836607056186159104/1524838665",
                "thumbs:down:count": 1,
                "thumbs:down:user_guids": Array [
                  "3",
                ],
                "thumbs:up:count": 2,
                "thumbs:up:user_guids": Array [
                  "1",
                  "2",
                ],
                "time_created": "1524838665",
                "time_published": "1524838665",
                "time_updated": "1524862542",
                "title": "Some Title",
                "type": "object",
                "wire_threshold": "",
              }
            }
            navigation={
              Object {
                "state": Object {
                  "params": Object {
                    "guid": 1,
                  },
                },
              }
            }
            rightToolbar={
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                    },
                    Object {
                      "paddingLeft": 10,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                      "padding": 4,
                    }
                  }
                >
                  <RemindAction
                    entity={
                      BlogModel {
                        "access_id": "2",
                        "boost_rejection_reason": -1,
                        "categories": Array [],
                        "category": "education",
                        "container_guid": "100",
                        "custom_meta": Object {
                          "author": "",
                          "description": "",
                          "title": "",
                        },
                        "description": "Some Blog",
                        "draft_access_id": "0",
                        "excerpt": "Some description",
                        "featured": false,
                        "featured_id": false,
                        "guid": "1",
                        "header_bg": "1",
                        "header_top": "0",
                        "last_save": "1524843907",
                        "last_updated": "1524838665",
                        "license": "attribution-noncommercial-cc",
                        "mature": false,
                        "monetized": false,
                        "ownerObj": UserModel {
                          "access_id": "2",
                          "banned": "no",
                          "blocked": undefined,
                          "boostProPlus": false,
                          "boost_autorotate": true,
                          "boost_rating": "2",
                          "briefdescription": "",
                          "categories": Array [
                            "music",
                            "nature",
                            "news",
                            "politics",
                            "science",
                            "technology",
                            "education",
                            "art",
                            "health",
                          ],
                          "chat": true,
                          "city": "",
                          "container_guid": "0",
                          "deleted": "0",
                          "disabled_boost": false,
                          "dob": "",
                          "eth_wallet": "0x6fffffdadae350ddd5a1777d5a1777a4f39f0317",
                          "fb": false,
                          "feature_flags": false,
                          "featured_id": "725165605782560768",
                          "founder": false,
                          "gender": "private",
                          "getAvatarSource": [MockFunction],
                          "guid": 100,
                          "icontime": "1523515420",
                          "language": "en",
                          "legacy_guid": false,
                          "mature": "1",
                          "mature_channel": "0",
                          "mature_visibility": false,
                          "merchant": Object {
                            "exclusive": Object {
                              "amount": "10",
                              "enabled": true,
                            },
                            "id": "acct_Dsdfjdlsaf34243",
                            "service": "stripe",
                          },
                          "monetized": "",
                          "name": "Some User",
                          "owner_guid": "0",
                          "pinned_posts": Array [],
                          "plus": true,
                          "programs": Array [
                            "affiliate",
                          ],
                          "rating": "1",
                          "rewards": true,
                          "signup_method": false,
                          "site_guid": false,
                          "social_profiles": Array [],
                          "spam": "0",
                          "subscribed": false,
                          "subscriber": true,
                          "subtype": false,
                          "time_created": "1468113204",
                          "time_updated": false,
                          "type": "user",
                          "username": "someusermane",
                          "verified": true,
                          "website": "",
                          "wire_rewards": Object {
                            "description": "",
                            "rewards": Object {
                              "money": Array [],
                              "points": Array [],
                            },
                          },
                        },
                        "owner_guid": "1",
                        "paywall": false,
                        "perma_url": "https://www.minds.com/blog/view/836607056186159104",
                        "published": true,
                        "rating": 1,
                        "reminds": 58,
                        "route": "blog/view/836607056186159104",
                        "slug": "",
                        "subtype": "blog",
                        "thumbnail_src": "https://www.minds.com/fs/v1/banners/836607056186159104/1524838665",
                        "thumbs:down:count": 1,
                        "thumbs:down:user_guids": Array [
                          "3",
                        ],
                        "thumbs:up:count": 2,
                        "thumbs:up:user_guids": Array [
                          "1",
                          "2",
                        ],
                        "time_created": "1524838665",
                        "time_published": "1524838665",
                        "time_updated": "1524862542",
                        "title": "Some Title",
                        "type": "object",
                        "wire_threshold": "",
                      }
                    }
                    navigation={
                      Object {
                        "state": Object {
                          "params": Object {
                            "guid": 1,
                          },
                        },
                      }
                    }
                    size={16}
                  />
                  <ThumbUpAction
                    entity={
                      BlogModel {
                        "access_id": "2",
                        "boost_rejection_reason": -1,
                        "categories": Array [],
                        "category": "education",
                        "container_guid": "100",
                        "custom_meta": Object {
                          "author": "",
                          "description": "",
                          "title": "",
                        },
                        "description": "Some Blog",
                        "draft_access_id": "0",
                        "excerpt": "Some description",
                        "featured": false,
                        "featured_id": false,
                        "guid": "1",
                        "header_bg": "1",
                        "header_top": "0",
                        "last_save": "1524843907",
                        "last_updated": "1524838665",
                        "license": "attribution-noncommercial-cc",
                        "mature": false,
                        "monetized": false,
                        "ownerObj": UserModel {
                          "access_id": "2",
                          "banned": "no",
                          "blocked": undefined,
                          "boostProPlus": false,
                          "boost_autorotate": true,
                          "boost_rating": "2",
                          "briefdescription": "",
                          "categories": Array [
                            "music",
                            "nature",
                            "news",
                            "politics",
                            "science",
                            "technology",
                            "education",
                            "art",
                            "health",
                          ],
                          "chat": true,
                          "city": "",
                          "container_guid": "0",
                          "deleted": "0",
                          "disabled_boost": false,
                          "dob": "",
                          "eth_wallet": "0x6fffffdadae350ddd5a1777d5a1777a4f39f0317",
                          "fb": false,
                          "feature_flags": false,
                          "featured_id": "725165605782560768",
                          "founder": false,
                          "gender": "private",
                          "getAvatarSource": [MockFunction],
                          "guid": 100,
                          "icontime": "1523515420",
                          "language": "en",
                          "legacy_guid": false,
                          "mature": "1",
                          "mature_channel": "0",
                          "mature_visibility": false,
                          "merchant": Object {
                            "exclusive": Object {
                              "amount": "10",
                              "enabled": true,
                            },
                            "id": "acct_Dsdfjdlsaf34243",
                            "service": "stripe",
                          },
                          "monetized": "",
                          "name": "Some User",
                          "owner_guid": "0",
                          "pinned_posts": Array [],
                          "plus": true,
                          "programs": Array [
                            "affiliate",
                          ],
                          "rating": "1",
                          "rewards": true,
                          "signup_method": false,
                          "site_guid": false,
                          "social_profiles": Array [],
                          "spam": "0",
                          "subscribed": false,
                          "subscriber": true,
                          "subtype": false,
                          "time_created": "1468113204",
                          "time_updated": false,
                          "type": "user",
                          "username": "someusermane",
                          "verified": true,
                          "website": "",
                          "wire_rewards": Object {
                            "description": "",
                            "rewards": Object {
                              "money": Array [],
                              "points": Array [],
                            },
                          },
                        },
                        "owner_guid": "1",
                        "paywall": false,
                        "perma_url": "https://www.minds.com/blog/view/836607056186159104",
                        "published": true,
                        "rating": 1,
                        "reminds": 58,
                        "route": "blog/view/836607056186159104",
                        "slug": "",
                        "subtype": "blog",
                        "thumbnail_src": "https://www.minds.com/fs/v1/banners/836607056186159104/1524838665",
                        "thumbs:down:count": 1,
                        "thumbs:down:user_guids": Array [
                          "3",
                        ],
                        "thumbs:up:count": 2,
                        "thumbs:up:user_guids": Array [
                          "1",
                          "2",
                        ],
                        "time_created": "1524838665",
                        "time_published": "1524838665",
                        "time_updated": "1524862542",
                        "title": "Some Title",
                        "type": "object",
                        "wire_threshold": "",
                      }
                    }
                    me={
                      Object {
                        "guid": "guidguid",
                      }
                    }
                    orientation="column"
                    size={16}
                  />
                  <ThumbDownAction
                    entity={
                      BlogModel {
                        "access_id": "2",
                        "boost_rejection_reason": -1,
                        "categories": Array [],
                        "category": "education",
                        "container_guid": "100",
                        "custom_meta": Object {
                          "author": "",
                          "description": "",
                          "title": "",
                        },
                        "description": "Some Blog",
                        "draft_access_id": "0",
                        "excerpt": "Some description",
                        "featured": false,
                        "featured_id": false,
                        "guid": "1",
                        "header_bg": "1",
                        "header_top": "0",
                        "last_save": "1524843907",
                        "last_updated": "1524838665",
                        "license": "attribution-noncommercial-cc",
                        "mature": false,
                        "monetized": false,
                        "ownerObj": UserModel {
                          "access_id": "2",
                          "banned": "no",
                          "blocked": undefined,
                          "boostProPlus": false,
                          "boost_autorotate": true,
                          "boost_rating": "2",
                          "briefdescription": "",
                          "categories": Array [
                            "music",
                            "nature",
                            "news",
                            "politics",
                            "science",
                            "technology",
                            "education",
                            "art",
                            "health",
                          ],
                          "chat": true,
                          "city": "",
                          "container_guid": "0",
                          "deleted": "0",
                          "disabled_boost": false,
                          "dob": "",
                          "eth_wallet": "0x6fffffdadae350ddd5a1777d5a1777a4f39f0317",
                          "fb": false,
                          "feature_flags": false,
                          "featured_id": "725165605782560768",
                          "founder": false,
                          "gender": "private",
                          "getAvatarSource": [MockFunction],
                          "guid": 100,
                          "icontime": "1523515420",
                          "language": "en",
                          "legacy_guid": false,
                          "mature": "1",
                          "mature_channel": "0",
                          "mature_visibility": false,
                          "merchant": Object {
                            "exclusive": Object {
                              "amount": "10",
                              "enabled": true,
                            },
                            "id": "acct_Dsdfjdlsaf34243",
                            "service": "stripe",
                          },
                          "monetized": "",
                          "name": "Some User",
                          "owner_guid": "0",
                          "pinned_posts": Array [],
                          "plus": true,
                          "programs": Array [
                            "affiliate",
                          ],
                          "rating": "1",
                          "rewards": true,
                          "signup_method": false,
                          "site_guid": false,
                          "social_profiles": Array [],
                          "spam": "0",
                          "subscribed": false,
                          "subscriber": true,
                          "subtype": false,
                          "time_created": "1468113204",
                          "time_updated": false,
                          "type": "user",
                          "username": "someusermane",
                          "verified": true,
                          "website": "",
                          "wire_rewards": Object {
                            "description": "",
                            "rewards": Object {
                              "money": Array [],
                              "points": Array [],
                            },
                          },
                        },
                        "owner_guid": "1",
                        "paywall": false,
                        "perma_url": "https://www.minds.com/blog/view/836607056186159104",
                        "published": true,
                        "rating": 1,
                        "reminds": 58,
                        "route": "blog/view/836607056186159104",
                        "slug": "",
                        "subtype": "blog",
                        "thumbnail_src": "https://www.minds.com/fs/v1/banners/836607056186159104/1524838665",
                        "thumbs:down:count": 1,
                        "thumbs:down:user_guids": Array [
                          "3",
                        ],
                        "thumbs:up:count": 2,
                        "thumbs:up:user_guids": Array [
                          "1",
                          "2",
                        ],
                        "time_created": "1524838665",
                        "time_published": "1524838665",
                        "time_updated": "1524862542",
                        "title": "Some Title",
                        "type": "object",
                        "wire_threshold": "",
                      }
                    }
                    me={
                      Object {
                        "guid": "guidguid",
                      }
                    }
                    orientation="column"
                    size={16}
                  />
                </View>
              </View>
            }
          >
            <Text
              style={
                Object {
                  "color": "#888",
                  "fontSize": 11,
                }
              }
            >
              Apr 27, 2018, 07:17
            </Text>
          </OwnerBlock>
        </View>
        <View
          style={
            Object {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          <BlogViewHTML
            html="Some Blog"
          />
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "padding": 12,
            }
          }
        >
          <Icon
            color="#b0bec5"
            name="public"
            onPress={[Function]}
            raised={false}
            reverse={false}
            reverseColor="white"
            size={18}
            underlayColor="white"
          />
          <Text
            style={
              Array [
                Object {
                  "fontSize": 10,
                },
                Object {
                  "paddingLeft": 5,
                },
                Object {
                  "color": "#b0bec5",
                },
                Object {
                  "paddingRight": 10,
                },
              ]
            }
          >
            Attribution-NonCommerical CC BY-NC
          </Text>
          <Icon
            color="#4690D6"
            name="share"
            onPress={[Function]}
            raised={false}
            reverse={false}
            reverseColor="white"
            size={20}
            underlayColor="white"
          />
        </View>
        <SafeAreaView
          style={
            Object {
              "left": 0,
              "position": "absolute",
              "top": 0,
            }
          }
        >
          <Icon
            color="#4690D6"
            name="arrow-back"
            onPress={[Function]}
            raised={true}
            reverse={false}
            reverseColor="white"
            size={22}
            underlayColor="white"
          />
        </SafeAreaView>
      </View>
    }
    navigation={
      Object {
        "state": Object {
          "params": Object {
            "guid": 1,
          },
        },
      }
    }
    store={
      CommentsStore {
        "attachment": AttachmentStore {
          "guid": "",
          "tempIosVideo": "",
        },
        "embed": RichEmbedStore {
          "_richEmbedFetchTimer": null,
          "clearRichEmbedAction": [Function],
          "richEmbedUrl": "",
          "setRichEmbedPromise": null,
        },
        "entity": null,
        "focusedUrn": null,
        "guid": "",
        "level": 0,
        "loadNext": "",
        "loadPrevious": "",
        "onAttachedMedia": [Function],
        "parent": null,
        "reversed": true,
        "selectMediaType": [Function],
        "socketRoomName": "",
      }
    }
  />
</View>
`;
